### DISCUSSION TITLE
Howto dynamic Or Where on same Attribute?

### DISCUSSION THREAD

**Discussion by t0mcat1337** [DirectoryTree/LdapRecord#382] [Q&A] [answered]
Hi everybody,

I want to build a LDAP Query based on a dynamic number of input values (as an array?) as an OR of the same LDAP Attribute (lets say a uid).

Examples:
input=["UserID1"];
should lead to a LDAP Filter like this:
filter="(uid=UserID1)"

input=["UserID1","UserID2"];
should lead to a LDAP Filter like this:
filter="(|(uid=UserID1)(uid=UserID2))"

input=["UserID1","UserID2","UserID3"];
should lead to a LDAP Filter like this:
filter="(|(uid=UserID1)(uid=UserID2)(uid=UserID3))"

As you can see, the number of input values is dynamic, so it is not possible to use queryBuilders "$query->orWhere()" methods (https://ldaprecord.com/docs/core/v2/searching/#or-wheres) as it is impossible to have a dynamic number of them in the code.

I can't find a method like "whereIn", which could accept an array of values and then build an Or Filter of them like this:
$query->whereIn("uid",["UserID1","UserID2","UserID3"])
==> "(|(uid=UserID1)(uid=UserID2)(uid=UserID3))"

At the moment I just can use "$query->rawFilter" and build the filter on my own to achieve this, correct? 
Or have I missed sth?

Thx in advance
T0mcat

**Comment by stevebauman** [comment, marked as answer]
Hi @t0mcat1337,

Would this work for you?

```php
use LdapRecord\Connection;

$conn = new Connection();

$query = $conn->query();

$input = ["UserID1","UserID2","UserID3"];

foreach ($input as $id) {
    $query->orWhere('uid', '=', $id);
}

// "(|(uid=UserID1)(uid=UserID2)(uid=UserID3))"
echo $query->getUnescapedQuery();
```

**Comment by t0mcat1337** [comment]
@stevebauman : 
Thx a lot for putting me to this. Didn't really get that "orWhere" can also be used sequentially and doesn't have to be chained together in one query - codeline.

**Reply by stevebauman** [reply]
Happy to help @t0mcat1337! Please drop by again if you have any further questions 😄
